### PR TITLE
JIT: Remove old loop code

### PR DESCRIPTION
**Pull request by jakobbotsch**
All dependencies on old loop finding have been removed, so remove all old code associated with it.

No diffs expected. Some TP improvements expected from removing old loop finding and from removing a DFS of the flow graph.

### CODE REVIEW

**Comment by BruceForstall** [comment]
[Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=533835&view=ms.vss-build-web.run-extensions-tab)

No asm diffs; significant TP improvements.

File: src/coreclr/jit/fgdiagnostic.cpp [hunk 1]
```diff
@@ -4737,412 +4677,18 @@ void Compiler::fgDebugCheckSsa()
 //
 void Compiler::fgDebugCheckLoopTable()
 {
-    if ((m_loops != nullptr) && optLoopsRequirePreHeaders)
+    if (m_loops == nullptr)
```
Comment (BruceForstall) [inline src/coreclr/jit/fgdiagnostic.cpp:4680]: Looks like the header comment above needs to be updated?

**Review by BruceForstall — APPROVED** [review]
Congrats on getting this far so quickly!

File: src/coreclr/jit/fgdiagnostic.cpp (hunk 1 above)
Comment (jakobbotsch) [reply src/coreclr/jit/fgdiagnostic.cpp:4664]: Done. Also rephrased a few other things (removed references to "loop table") and deleted the loop flag enum that had been left over.

**Comment by jakobbotsch** [comment]
cc @dotnet/jit-contrib

**Comment by jakobbotsch** [comment]
> Congrats on getting this far so quickly!

Thanks! There's still some regressions to investigate, but overall I'm happy with how this ended up, and now I have a very good understanding of how all the pieces fit together :-)

**Pull request merged**